neps/nep-0393.md

```diff
@@ -0,0 +1,480 @@
+
+    Note over Alice,SBT_Registry: now Alice can query registry to check her SBT
+
+    Alice-->>SBT_Registry: sbt(SBT_1_Contract, 238)
+    SBT_Registry-->>Alice: {token: 238, owner: alice, metadata}
+
+    Note over Alice,SBT_Registry: Alice participates in other community and applies for another SBT.<br/> She will get 2 SBTs. However she uses a different wallet: alice2.
+
+    participant SBT2 as SBT_2 Contract
+    actor Issuer2 as SBT_2 Issuer
+
+    Issuer2->>SBT2: sbt_mint_multi([[alice2, metadata2], [alice2, metadata3]])
+    activate SBT2
+    SBT2-)SBT_Registry: sbt_mint([[alice2, metadata2], [alice2, metadata3]])
+    SBT_Registry->>SBT_Registry: emit Mint(SBT_2_Contract, alice2, [7991, 17992])
+    SBT_Registry-)SBT2: [7991, 1992]
+    deactivate SBT2
+
+    Alice-->>SBT_Registry: sbt(SBT_2_Contract, 7991)
+    SBT_Registry-->>Alice: {token: 7991, owner: alice2, metadata: metadata2}
+
+    Note over Alice,SBT_Registry: After a while Alice decides to merge her `alice2` account into `alice2`
+    Alice->>SBT_Registry: sbt_soul_transfer(alice) --accountId alice2
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice2)
+    SBT_Registry-->>-Alice: []
+
+    Alice-->>+SBT_Registry: sbt_tokens_by_owner(alice)
+    SBT_Registry-->>-Alice: [[SBT_1_Contract, [238]], [SBT_2_Contract, [7991, 7992]]]}
+```
+
+## Consequences
+
+Being fully compatible with NFT standard is a desirable. However, given the requirements related to _soul transfer_ we didn't find an applaudable solution. Also we decided to use u64 as a Token ID, diverging further from the NFT NEP-171 standard.
+
+Give that our requirements are much striker, we had to reconsider the level of compatibility with NEP-171 NFT.
+There are many examples where NFT standards are poorly or improperly implemented, adding another standard with different functionality but equal naming will cause lots of problems and misclassifications between NFT and SBT.
+
+### Positive
+
+- Template and set of guidelines for creating SBT tokens.
+- Ability to create SBT aggregators.
+- Ability to use SBT as a primitive to model non KYC identity, badges, certificates etc...
+- SBT can be further used for "lego" protocols, like: Proof of Humanity (discussed for NDC Governance), undercollateralized lending, role based authentication systems, innovative economic and social applications...
+- Standard recoverability mechanism.
+- SBT are considered as a basic primitive for Decentralized Societies.
+- new way to implement Sybil attack resistance.
+
+### Neutral
+
+- The API partially follows the NEP-171 (NFT) standard. The proposed design is to have native SBT API and make it possible for issuer contracts to support NFT based queries if needed (such contract will have a limitation of only issuing SBTs with one `ClassId` only).
+
+### Negative
+
+- New set of events to be handled by the indexer and wallets.
+- Complexity of integration with a registry: all SBT related transactions must go through Registry.
```

Comment:
I find the lack of consideration for spam concerning. In this standard, there would inevitably be open registries. Using such registries it would be trivial to create thousands of tokens in an account, that they would have no recourse against.

Reply:
You are referring that an issuer issue SBTs to many wallets and doxxing account identity. This is a case of any public, permissionless, not fully private blockchian. There are multiple ways how someone can spam a account today:
* Create FT, block transfers (panic), and issue them to multiple accounts
* do the same with NFT
* do the same with any other primitive

Similarly, there are many ways how someone can use any media to doxx identity: making transactions with memo content about account identity, publishing a spreadsheet with associated personal data (IPFS, Tweets, Google spreadsheet....).

Registry has the `Burn` event, and it's up to the registry how to handle it. Accordingly, FT and NFT have a Burn event, but doesn't have standard function for that.

Token standard can't solve it.

Reply:
> Create FT, block transfers (panic), and issue them to multiple accounts

This does not adhere to the NEP-141 standard, and wallets should rightfully ignore it.

> do the same with NFT

Same. Not standard compliant. Wallets should ignore them. Spam SBTs would be compliant to the standard.

Also, you are confusing two different problems. Doxxing, and spam. The two are very different concepts here. Yes, permanent doxxing is unavoidable in a public blockchain. Spam is not.

The token standard could solve this, if this was a priority, by having the user consent to their SBTs, otherwise hiding them/marking them as not valid.

Reply:
I don't understand why do you claim that spam NFT are not standard compliant and spam SBTs are.

Reply:
If an NFT cannot be transferred, it does not adhere to the core of NEP-171. Thus, if wallets detect that an NFT cannot be transferred by their owner, they should rightfully start ignoring that NFT. TL;DR: An NFT that cannot be transferred is not an NEP-171 compliant NFT.
The same cannot be said about SBTs.